### PR TITLE
feat(storage): introduce non-proto type for vnode bitmap

### PR DESCRIPTION
**Pull request by xx01cyx**
### Summarize your change 

Introduce non-proto type for vnode bitmap.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)

#2882 should depend on this PR. More specifically, the non-proto type introduced by this PR will be used in keyspace and interfaces of state store (get, scan, etc.).

### CODE REVIEW

File: src/common/src/consistent_hash/vnode.rs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    }
+
+    pub fn check_overlap(&self, bitmaps: &[risingwave_pb::common::VNodeBitmap]) -> bool {
+        if bitmaps.is_empty() {
+            return true;
+        }
+        if let Ok(pos) =
+            bitmaps.binary_search_by_key(&self.table_id, |bitmap| bitmap.get_table_id())
+        {
+            let text = &bitmaps[pos];
```
Comment (fuyufjh) [inline src/common/src/consistent_hash/vnode.rs:51]: a small question: why it's called "text"? 👀

File: src/common/src/consistent_hash/vnode.rs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+pub type VirtualNode = u16;
+pub const VNODE_BITS: usize = 11;
+pub const VIRTUAL_NODE_COUNT: usize = 1 << VNODE_BITS;
+pub const VNODE_BITMAP_LEN: usize = 1 << (VNODE_BITS - 3);
+
+// `VNodeBitmap` is a bitmap of vnodes for a state table, which indicates the vnodes that the state
+// table owns.
+#[derive(Clone, Default)]
+pub struct VNodeBitmap {
+    table_id: u32,
```
Comment (fuyufjh) [inline src/common/src/consistent_hash/vnode.rs:26]: Looks a little strange. Are you sure every `VNodeBitmap` needs a `table_id`?

**Review by fuyufjh — APPROVED** [review]
Generally LGTM

File: src/common/src/consistent_hash/vnode.rs (hunk 1 above)
Comment (xx01cyx) [reply src/common/src/consistent_hash/vnode.rs:51]: Just copied the function from `store.rs`. 🤣 Weird indeed. Lemme change it.

File: src/common/src/consistent_hash/vnode.rs (hunk 2 above)
Comment (xx01cyx) [reply src/common/src/consistent_hash/vnode.rs:26]: There will be a `VNodeBitmap` in every `Keyspace` of a state table, so `table_id` is in need here. Also, `VNodeBitmap` will be added to the interface of `StateStore`, where we use it for read pruning. In this case, `table_id` is also necessary. Any other case to use `VNodeBitmap`?

Comment (xx01cyx) [reply src/common/src/consistent_hash/vnode.rs:26]: And from the aspect of logic, the struct is designed for read pruning, which depends on state table. So having a `table_id` is reasonable, right? 🤔

Comment (fuyufjh) [reply src/common/src/consistent_hash/vnode.rs:26]: Sounds reasonable 😁

**Pull request merged**